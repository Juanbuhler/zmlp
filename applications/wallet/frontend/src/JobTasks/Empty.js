import { typography } from '../Styles'

import NoJobsSvg from '../Icons/noJobs.svg'

const JobTasksEmpty = () => {
  return (
    <div
      css={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        fontSize: typography.size.large,
        lineHeight: typography.height.large,
      }}
    >
      <NoJobsSvg width={200} />
      <div>This job has no tasks.</div>
      <div>Any new task will appear here.</div>
    </div>
  )
}

export default JobTasksEmpty
