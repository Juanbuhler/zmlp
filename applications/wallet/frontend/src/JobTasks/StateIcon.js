import PropTypes from 'prop-types'

import { colors } from '../Styles'

import CrossSmallSvg from '../Icons/crossSmall.svg'
import ClockSvg from '../Icons/clock.svg'
import SquareSquareSquareSvg from '../Icons/squareSquareSquare.svg'
import PausedSvg from '../Icons/paused.svg'
import CheckmarkSvg from '../Icons/checkmark.svg'
import GeneratingSvg from '../Icons/generating.svg'

const ICON_SIZE = 20

const JobTasksStateIcon = ({ state }) => {
  switch (state) {
    case 'Waiting':
    case 'Depend':
      return <ClockSvg color={colors.structure.white} width={ICON_SIZE} />

    case 'Running':
      return (
        <GeneratingSvg color={colors.signal.canary.base} width={ICON_SIZE} />
      )

    case 'Success':
      return <CheckmarkSvg color={colors.signal.grass.base} width={ICON_SIZE} />

    case 'Skipped':
      return <PausedSvg color={colors.structure.steel} width={ICON_SIZE} />

    case 'Queued':
      return (
        <SquareSquareSquareSvg
          color={colors.signal.sky.base}
          width={ICON_SIZE}
        />
      )

    case 'Failure':
    default:
      return (
        <CrossSmallSvg color={colors.signal.warning.base} width={ICON_SIZE} />
      )
  }
}

JobTasksStateIcon.propTypes = {
  state: PropTypes.oneOf([
    'Waiting',
    'Depend',
    'Running',
    'Success',
    'Failure',
    'Skipped',
    'Queued',
  ]).isRequired,
}

export default JobTasksStateIcon
