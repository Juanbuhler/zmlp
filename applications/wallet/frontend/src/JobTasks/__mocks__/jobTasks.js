const jobTasks = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 'c812fa09-5f3a-11ea-bb46-6a6895b1a9f6',
      jobId: 'ea7836f1-44a7-19e1-a67f-1e6fa1f82545',
      projectId: '00000000-0000-0000-0000-000000000000',
      dataSourceId: 'bf855728-5f3a-11ea-bb46-6a6895b1a9f6',
      name: 'Expand with 20 assets, 8 processors.',
      state: 'Waiting',
      host: 'http://analyst-7b7fd7958-9zl69:5000',
      timeStarted: 1583451666243,
      timeStopped: 1583451715081,
      timeCreated: 1583451665765,
      timePing: 1583451688085,
      assetCounts: {
        assetCreatedCount: 0,
        assetReplacedCount: 20,
        assetWarningCount: 0,
        assetErrorCount: 0,
        assetTotalCount: 20,
      },
      taskId: 'c812fa09-5f3a-11ea-bb46-6a6895b1a9f6',
      url:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/00000000-0000-0000-0000-000000000000/tasks/c812fa09-5f3a-11ea-bb46-6a6895b1a9f6/',
    },
    {
      id: 'ea7836f2-44a7-19e1-a67f-1e6fa1f82545',
      jobId: 'ea7836f1-44a7-19e1-a67f-1e6fa1f82545',
      projectId: '00000000-0000-0000-0000-000000000000',
      dataSourceId: 'bf855728-5f3a-11ea-bb46-6a6895b1a9f6',
      name:
        "Crawling files in 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/cats/'",
      state: 'Running',
      host: 'http://analyst-7b7fd7958-9zl69:5000',
      timeStarted: 1583451662066,
      timeStopped: 1583451666124,
      timeCreated: 1583451652154,
      timePing: 1583451652154,
      assetCounts: {
        assetCreatedCount: 35,
        assetReplacedCount: 0,
        assetWarningCount: 0,
        assetErrorCount: 0,
        assetTotalCount: 0,
      },
      taskId: 'ea7836f2-44a7-19e1-a67f-1e6fa1f82545',
      url:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/00000000-0000-0000-0000-000000000000/tasks/ea7836f2-44a7-19e1-a67f-1e6fa1f82545/',
    },
    {
      id: 'ea7836f3-44a7-19e1-a67f-1e6fa1f82545',
      jobId: 'ea7836f1-44a7-19e1-a67f-1e6fa1f82545',
      projectId: '00000000-0000-0000-0000-000000000000',
      dataSourceId: 'bf855728-5f3a-11ea-bb46-6a6895b1a9f6',
      name: 'Expand with 15 assets, 8 processors.',
      state: 'Success',
      host: 'http://analyst-7b7fd7958-9zl69:5000',
      timeStarted: 1583451715229,
      timeStopped: 1583451752821,
      timeCreated: 1583451665950,
      timePing: 1583451748150,
      assetCounts: {
        assetCreatedCount: 0,
        assetReplacedCount: 15,
        assetWarningCount: 0,
        assetErrorCount: 0,
        assetTotalCount: 15,
      },
      taskId: 'ea7836f3-44a7-19e1-a67f-1e6fa1f82545',
      url:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/00000000-0000-0000-0000-000000000000/tasks/ea7836f3-44a7-19e1-a67f-1e6fa1f82545/',
    },
    {
      id: 'c812fa09-5f3a-11ea-bb46-6a6895b1a9f9',
      jobId: 'ea7836f1-44a7-19e1-a67f-1e6fa1f82545',
      projectId: '00000000-0000-0000-0000-000000000000',
      dataSourceId: 'bf855728-5f3a-11ea-bb46-6a6895b1a9f6',
      name: 'Expand with 20 assets, 8 processors.',
      state: 'Failure',
      host: 'http://analyst-7b7fd7958-9zl69:5000',
      timeStarted: 1583451666243,
      timeStopped: 1583451715081,
      timeCreated: 1583451665765,
      timePing: 1583451688085,
      assetCounts: {
        assetCreatedCount: 0,
        assetReplacedCount: 20,
        assetWarningCount: 0,
        assetErrorCount: 0,
        assetTotalCount: 20,
      },
      taskId: 'c812fa09-5f3a-11ea-bb46-6a6895b1a9f9',
      url:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/00000000-0000-0000-0000-000000000000/tasks/c812fa09-5f3a-11ea-bb46-6a6895b1a9f9/',
    },
    {
      id: 'ea7836f2-44a7-19e1-a67f-1e6fa1f82549',
      jobId: 'ea7836f1-44a7-19e1-a67f-1e6fa1f82545',
      projectId: '00000000-0000-0000-0000-000000000000',
      dataSourceId: 'bf855728-5f3a-11ea-bb46-6a6895b1a9f6',
      name:
        "Crawling files in 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/cats/'",
      state: 'Skipped',
      host: 'http://analyst-7b7fd7958-9zl69:5000',
      timeStarted: 1583451662066,
      timeStopped: 1583451666124,
      timeCreated: 1583451652154,
      timePing: 1583451652154,
      assetCounts: {
        assetCreatedCount: 35,
        assetReplacedCount: 0,
        assetWarningCount: 0,
        assetErrorCount: 0,
        assetTotalCount: 0,
      },
      taskId: 'ea7836f2-44a7-19e1-a67f-1e6fa1f82549',
      url:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/00000000-0000-0000-0000-000000000000/tasks/ea7836f2-44a7-19e1-a67f-1e6fa1f825459/',
    },
    {
      id: 'ea7836f3-44a7-19e1-a67f-1e6fa1f82549',
      jobId: 'ea7836f1-44a7-19e1-a67f-1e6fa1f82545',
      projectId: '00000000-0000-0000-0000-000000000000',
      dataSourceId: 'bf855728-5f3a-11ea-bb46-6a6895b1a9f6',
      name: 'Expand with 15 assets, 8 processors.',
      state: 'Queued',
      host: 'http://analyst-7b7fd7958-9zl69:5000',
      timeStarted: 1583451715229,
      timeStopped: 1583451752821,
      timeCreated: 1583451665950,
      timePing: 1583451748150,
      assetCounts: {
        assetCreatedCount: 0,
        assetReplacedCount: 15,
        assetWarningCount: 0,
        assetErrorCount: 0,
        assetTotalCount: 15,
      },
      taskId: 'ea7836f3-44a7-19e1-a67f-1e6fa1f82549',
      url:
        'https://wallet.zmlp.zorroa.com/api/v1/projects/00000000-0000-0000-0000-000000000000/tasks/ea7836f3-44a7-19e1-a67f-1e6fa1f82549/',
    },
  ],
}

export default jobTasks
