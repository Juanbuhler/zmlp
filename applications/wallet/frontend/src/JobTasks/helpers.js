export const getDuration = ({ timeStarted, timeStopped, now }) => {
  const stopTime = timeStopped < 0 ? now : timeStopped
  return stopTime - timeStarted
}
