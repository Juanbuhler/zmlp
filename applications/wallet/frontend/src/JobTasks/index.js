import { useRouter } from 'next/router'

import Table from '../Table'

import JobTasksEmpty from './Empty'
import JobTasksRow from './Row'

const JobTasks = () => {
  const {
    query: { projectId, jobId },
  } = useRouter()

  return (
    <Table
      assetType="Tasks"
      url={`/api/v1/projects/${projectId}/jobs/${jobId}/tasks`}
      columns={[
        'Task State',
        'Task ID',
        'Description',
        'Duration',
        'Assets',
        'Updated',
        'Errors',
      ]}
      expandColumn={3}
      renderEmpty={<JobTasksEmpty />}
      renderRow={({ result }) => <JobTasksRow key={result.id} task={result} />}
    />
  )
}

export default JobTasks
