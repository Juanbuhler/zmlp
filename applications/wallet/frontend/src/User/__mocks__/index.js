import { createContext, createElement } from 'react'

import user from './user'

export const noop = () => {}

export const UserContext = createContext({
  user,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ children, ...rest }) => createElement('User', rest, children)

export default User
