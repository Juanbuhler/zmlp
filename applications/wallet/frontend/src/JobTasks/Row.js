import PropTypes from 'prop-types'
import Router from 'next/router'
import Link from 'next/link'

import { formatFullDate, getDuration, formatDuration } from '../Date/helpers'
import { spacing, typography } from '../Styles'

import JobTasksStateIcon from './StateIcon'

const JobTasksRow = ({
  projectId,
  jobId,
  task: {
    id: taskId,
    name,
    state,
    timeStarted,
    timeStopped,
    assetCounts: { assetErrorCount },
  },
}) => {
  const isStarted = timeStarted !== -1
  const taskDuration = getDuration({
    timeStarted,
    timeStopped,
    now: Date.now(),
  })

  return (
    <tr
      css={{ cursor: 'pointer' }}
      onClick={(event) => {
        const { target: { localName } = {} } = event || {}
        if (['a', 'button', 'svg', 'path'].includes(localName)) return
        Router.push(
          '/[projectId]/jobs/[jobId]/tasks/[taskId]/script',
          `/${projectId}/jobs/${jobId}/tasks/${taskId}/script`,
        )
      }}
    >
      <td css={{ display: 'flex' }}>
        <JobTasksStateIcon state={state} />
        <span css={{ paddingLeft: spacing.normal }}>{state}</span>
      </td>

      <td>
        <Link
          href="/[projectId]/jobs/[jobId]/tasks/[taskId]/script"
          as={`/${projectId}/jobs/${jobId}/tasks/${taskId}/script`}
          passHref
        >
          <a css={{ ':hover': { textDecoration: 'none' } }} title={taskId}>
            {taskId}
          </a>
        </Link>
      </td>

      <td title={name}>{name}</td>

      <td>
        {isStarted ? (
          formatFullDate({ timestamp: timeStarted })
        ) : (
          <div
            css={{ fontStyle: typography.style.italic }}
          >{`${state}...`}</div>
        )}
      </td>

      <td>
        {isStarted ? (
          formatDuration({
            seconds: taskDuration / 1000,
          })
        ) : (
          <div
            css={{ fontStyle: typography.style.italic }}
          >{`${state}...`}</div>
        )}
      </td>

      <td>{assetErrorCount}</td>
    </tr>
  )
}

JobTasksRow.propTypes = {
  projectId: PropTypes.string.isRequired,
  jobId: PropTypes.string.isRequired,
  task: PropTypes.shape({
    id: PropTypes.string.isRequired,
    jobId: PropTypes.string.isRequired,
    projectId: PropTypes.string.isRequired,
    dataSourceId: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    host: PropTypes.string.isRequired,
    timeStarted: PropTypes.number.isRequired,
    timeStopped: PropTypes.number.isRequired,
    timeCreated: PropTypes.number.isRequired,
    timePing: PropTypes.number.isRequired,
    assetCounts: PropTypes.shape({
      assetCreatedCount: PropTypes.number.isRequired,
      assetReplacedCount: PropTypes.number.isRequired,
      assetWarningCount: PropTypes.number.isRequired,
      assetErrorCount: PropTypes.number.isRequired,
      assetTotalCount: PropTypes.number.isRequired,
    }).isRequired,
    url: PropTypes.string.isRequired,
  }).isRequired,
}

export default JobTasksRow
