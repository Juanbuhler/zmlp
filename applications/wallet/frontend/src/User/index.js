import { createContext, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import * as Sentry from '@sentry/browser'

import { getUser, setUser } from './helpers'

export const noop = () => {}

export const UserContext = createContext({
  user: {},
  setUser: noop,
  googleAuth: {},
  setGoogleAuth: noop,
})

const User = ({ initialUser, children }) => {
  const [user, setStateUser] = useState(initialUser)
  const [hasLocalStorageLoaded, setHasLocalStorageLoaded] = useState(false)
  const [googleAuth, setGoogleAuth] = useState({ signIn: noop, signOut: noop })

  useEffect(() => {
    if (initialUser.id || hasLocalStorageLoaded) return

    const storedUser = getUser()

    setStateUser(storedUser)

    setHasLocalStorageLoaded(true)

    /* istanbul ignore next */
    Sentry.configureScope(scope => {
      scope.setUser(storedUser)
    })
  }, [initialUser, hasLocalStorageLoaded, user, setStateUser])

  if (!initialUser.id && !hasLocalStorageLoaded) return null

  return (
    <UserContext.Provider
      value={{
        user,
        setUser: setUser({ setStateUser, user }),
        googleAuth,
        setGoogleAuth,
      }}>
      {children}
    </UserContext.Provider>
  )
}

User.propTypes = {
  initialUser: PropTypes.shape({
    id: PropTypes.number,
    email: PropTypes.string,
  }).isRequired,
  children: PropTypes.node.isRequired,
}

export default User
