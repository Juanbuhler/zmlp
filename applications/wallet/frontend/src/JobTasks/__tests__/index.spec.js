import TestRenderer from 'react-test-renderer'

import jobTasks from '../__mocks__/jobTasks'
import mockUser from '../../User/__mocks__/user'

import User from '../../User'

import JobTasks from '..'

const PROJECT_ID = '76917058-b147-4556-987a-0a0f11e46d9b'
const JOB_ID = 'c097596f-62ef-1f81-83f8-0a580a000954'

jest.mock('../../Date/helpers', () => ({
  ...jest.requireActual('../../Date/helpers'),
  getDuration: () => 1587969769607,
}))

describe('<JobTasks />', () => {
  it('should render properly with job tasks', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/jobs/[jobId]/tasks',
      query: { projectId: PROJECT_ID, jobId: JOB_ID },
    })

    require('swr').__setMockUseSWRResponse({
      data: jobTasks,
    })

    const component = TestRenderer.create(
      <User initialUser={mockUser}>
        <JobTasks />
      </User>,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly without job tasks', () => {
    require('next/router').__setUseRouter({
      pathname: '/[projectId]/jobs/[jobId]/tasks',
      query: { projectId: PROJECT_ID, jobId: JOB_ID },
    })

    require('swr').__setMockUseSWRResponse({
      data: {},
    })

    const component = TestRenderer.create(
      <User initialUser={mockUser}>
        <JobTasks />
      </User>,
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
