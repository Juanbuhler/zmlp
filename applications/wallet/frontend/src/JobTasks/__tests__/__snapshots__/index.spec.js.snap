// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobTasks /> should render properly with job tasks 1`] = `
.emotion-26 {
  height: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 16px;
}

.emotion-0 {
  color: #b3b3b3;
  font-weight: 400;
}

.emotion-4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px 16px;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-4,
.emotion-4:hover,
.emotion-4:visited {
  background-color: #73b61c;
}

.emotion-4:hover {
  background-color: #96c955;
}

.emotion-4[aria-disabled=true] {
  background-color: #808080;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  padding-left: 4px;
}

.emotion-25 {
  width: 100%;
  height: auto;
  border-spacing: 0;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  white-space: nowrap;
}

.emotion-25 tr {
  background-color: #252525;
}

.emotion-25 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-25 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-25 tr:hover .gear {
  color: #b3b3b3;
}

.emotion-25 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-25 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-25 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-25 td {
  max-width: calc(100vw / 6);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-25 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-25 td:last-of-type {
  border-right: 1px solid transparent;
  overflow: visible;
}

.emotion-6 {
  text-align: left;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 12px 16px;
  border-bottom: 1px solid #2e2e2e;
}

.emotion-6:nth-of-type(3) {
  width: 100%;
}

.emotion-6:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-13 {
  padding-left: 16px;
}

<div
  className="emotion-26"
>
  <div
    className="emotion-5"
  >
    <h3
      className="emotion-0"
    >
      Number of 
      Tasks
      : 
      3
    </h3>
    <button
      className="emotion-4"
      onClick={[Function]}
      type="button"
    >
      <div
        className="emotion-3"
      >
        <svg
          className="emotion-1"
          color="#ffffff"
          data-file-name="refresh.svg"
          width={20}
        />
        <div
          className="emotion-2"
        >
          Refresh 
          Tasks
        </div>
      </div>
    </button>
  </div>
  <table
    className="emotion-25"
  >
    <thead>
      <tr>
        <th
          className="emotion-6"
        >
          Task State
        </th>
        <th
          className="emotion-6"
        >
          Task ID
        </th>
        <th
          className="emotion-6"
        >
          Description
        </th>
        <th
          className="emotion-6"
        >
          Duration
        </th>
        <th
          className="emotion-6"
        >
          Assets
        </th>
        <th
          className="emotion-6"
        >
          Updated
        </th>
        <th
          className="emotion-6"
        >
          Errors
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="emotion-14"
        >
          <svg
            color="#ffffff"
            data-file-name="clock.svg"
            width={20}
          />
          <span
            className="emotion-13"
          >
            Waiting
          </span>
        </td>
        <td
          title="c812fa09-5f3a-11ea-bb46-6a6895b1a9f6"
        >
          c812fa09-5f3a-11ea-bb46-6a6895b1a9f6
        </td>
        <td
          title="Expand with 20 assets, 8 processors."
        >
          Expand with 20 assets, 8 processors.
        </td>
        <td>
          13h : 33m : 58s
        </td>
        <td>
          20
        </td>
        <td>
          2020-03-05 23:41:28
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="emotion-14"
        >
          <svg
            color="#ebb52e"
            data-file-name="generating.svg"
            width={20}
          />
          <span
            className="emotion-13"
          >
            Running
          </span>
        </td>
        <td
          title="ea7836f2-44a7-19e1-a67f-1e6fa1f82545"
        >
          ea7836f2-44a7-19e1-a67f-1e6fa1f82545
        </td>
        <td
          title="Crawling files in 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/cats/'"
        >
          Crawling files in 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/cats/'
        </td>
        <td>
          01h : 07m : 38s
        </td>
        <td>
          0
        </td>
        <td>
          2020-03-05 23:40:52
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="emotion-14"
        >
          <svg
            color="#4ac565"
            data-file-name="checkmark.svg"
            width={20}
          />
          <span
            className="emotion-13"
          >
            Success
          </span>
        </td>
        <td
          title="ea7836f3-44a7-19e1-a67f-1e6fa1f82545"
        >
          ea7836f3-44a7-19e1-a67f-1e6fa1f82545
        </td>
        <td
          title="Expand with 15 assets, 8 processors."
        >
          Expand with 15 assets, 8 processors.
        </td>
        <td>
          10h : 26m : 32s
        </td>
        <td>
          15
        </td>
        <td>
          2020-03-05 23:42:28
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="emotion-14"
        >
          <svg
            color="#c92300"
            data-file-name="crossSmall.svg"
            width={20}
          />
          <span
            className="emotion-13"
          >
            Failure
          </span>
        </td>
        <td
          title="c812fa09-5f3a-11ea-bb46-6a6895b1a9f9"
        >
          c812fa09-5f3a-11ea-bb46-6a6895b1a9f9
        </td>
        <td
          title="Expand with 20 assets, 8 processors."
        >
          Expand with 20 assets, 8 processors.
        </td>
        <td>
          13h : 33m : 58s
        </td>
        <td>
          20
        </td>
        <td>
          2020-03-05 23:41:28
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="emotion-14"
        >
          <svg
            color="#808080"
            data-file-name="paused.svg"
            width={20}
          />
          <span
            className="emotion-13"
          >
            Skipped
          </span>
        </td>
        <td
          title="ea7836f2-44a7-19e1-a67f-1e6fa1f82549"
        >
          ea7836f2-44a7-19e1-a67f-1e6fa1f82549
        </td>
        <td
          title="Crawling files in 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/cats/'"
        >
          Crawling files in 'gs://zmlp-private-test-data/zorroa-deploy-testdata/zorroa-cypress-testdata/cats/'
        </td>
        <td>
          01h : 07m : 38s
        </td>
        <td>
          0
        </td>
        <td>
          2020-03-05 23:40:52
        </td>
        <td>
          0
        </td>
      </tr>
      <tr>
        <td
          className="emotion-14"
        >
          <svg
            color="#4a90e2"
            data-file-name="squareSquareSquare.svg"
            width={20}
          />
          <span
            className="emotion-13"
          >
            Queued
          </span>
        </td>
        <td
          title="ea7836f3-44a7-19e1-a67f-1e6fa1f82549"
        >
          ea7836f3-44a7-19e1-a67f-1e6fa1f82549
        </td>
        <td
          title="Expand with 15 assets, 8 processors."
        >
          Expand with 15 assets, 8 processors.
        </td>
        <td>
          10h : 26m : 32s
        </td>
        <td>
          15
        </td>
        <td>
          2020-03-05 23:42:28
        </td>
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
  <div>
     
  </div>
  <div>
     
  </div>
</div>
`;

exports[`<JobTasks /> should render properly without job tasks 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-top: 24px;
  padding-bottom: 16px;
}

.emotion-0 {
  color: #b3b3b3;
  font-weight: 400;
}

.emotion-4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 2px;
  margin: 0;
  padding: 12px 32px;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px 16px;
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4[aria-disabled=true] {
  color: #2e2e2e;
}

.emotion-4,
.emotion-4:hover,
.emotion-4:visited {
  background-color: #73b61c;
}

.emotion-4:hover {
  background-color: #96c955;
}

.emotion-4[aria-disabled=true] {
  background-color: #808080;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  padding-left: 4px;
}

.emotion-6 {
  text-align: left;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  color: #e0e0e0;
  background-color: #4a4a4a;
  padding: 12px 16px;
  border-bottom: 1px solid #2e2e2e;
}

.emotion-6:nth-of-type(3) {
  width: 100%;
}

.emotion-6:not(:last-child) {
  border-right: 1px solid #2e2e2e;
}

.emotion-17 {
  height: 100%;
}

.emotion-16 {
  width: 100%;
  height: 100%;
  border-spacing: 0;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.5);
  white-space: nowrap;
}

.emotion-16 tr {
  background-color: #252525;
}

.emotion-16 tr:nth-of-type(2n) {
  background-color: #2e2e2e;
}

.emotion-16 tr:hover {
  background-color: #4a4a4a;
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
}

.emotion-16 tr:hover .gear {
  color: #b3b3b3;
}

.emotion-16 tr:hover td {
  border: 1px solid #808080;
  border-left: 0;
  border-right: 0;
}

.emotion-16 tr:hover td:first-of-type {
  border-left: 1px solid #808080;
}

.emotion-16 tr:hover td:last-of-type {
  border-right: 1px solid #808080;
}

.emotion-16 td {
  max-width: calc(100vw / 6);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  color: #e0e0e0;
  padding: 8px 16px;
  border: 1px solid transparent;
  border-left: 0;
  border-right: 0;
}

.emotion-16 td:first-of-type {
  border-left: 1px solid transparent;
}

.emotion-16 td:last-of-type {
  border-right: 1px solid transparent;
  overflow: visible;
}

.emotion-15 {
  height: 100%;
  pointer-events: none;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  color: #808080;
  font-size: 15px;
  line-height: 19px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 18px;
  line-height: 21px;
}

<div
  className="emotion-17"
>
  <div
    className="emotion-5"
  >
    <h3
      className="emotion-0"
    >
      Number of 
      Tasks
      : 
      0
    </h3>
    <button
      className="emotion-4"
      onClick={[Function]}
      type="button"
    >
      <div
        className="emotion-3"
      >
        <svg
          className="emotion-1"
          color="#ffffff"
          data-file-name="refresh.svg"
          width={20}
        />
        <div
          className="emotion-2"
        >
          Refresh 
          Tasks
        </div>
      </div>
    </button>
  </div>
  <table
    className="emotion-16"
  >
    <thead>
      <tr>
        <th
          className="emotion-6"
        >
          Task State
        </th>
        <th
          className="emotion-6"
        >
          Task ID
        </th>
        <th
          className="emotion-6"
        >
          Description
        </th>
        <th
          className="emotion-6"
        >
          Duration
        </th>
        <th
          className="emotion-6"
        >
          Assets
        </th>
        <th
          className="emotion-6"
        >
          Updated
        </th>
        <th
          className="emotion-6"
        >
          Errors
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="emotion-15"
      >
        <td
          colSpan={7}
        >
          <div
            className="emotion-14"
          >
            <div
              className="emotion-13"
            >
              <svg
                data-file-name="noJobs.svg"
                width={200}
              />
              <div>
                This job has no tasks.
              </div>
              <div>
                Any new task will appear here.
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div>
     
  </div>
  <div>
     
  </div>
</div>
`;
